src/main/java/com/pnu/ordermanagementapp/product/ProductServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.pnu.ordermanagementapp.product;
+
+import com.pnu.ordermanagementapp.exception.ServiceException;
+import com.pnu.ordermanagementapp.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.*;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+
+    private ProductRepository productRepository;
+
+    private static int pageSize = 10;
```

Comment:
For these purposes use  @PageableDefault

Reply:
For now we don't accept `Pageable` as parameter in controller methods. We build it directly in service.
Hovewer, I believe, it can be refactored later.